File: genInp.py
import random
import sys
# define a set
s = set()
n = 1000000
N = str(n)

for i in range(1,n+1):
    # generate a number uniformly in [1, 400]
    x = random.randint(0, 10000)
    y = random.randint(0, 10000)
    # add the number to the set
    # check if the number is already in the set
    # if it is, generate another number
    while (x, y) in s:
        x = random.randint(0, 10000)
        y = random.randint(0, 10000)
    s.add((x, y))

with open('myInputs/inp1000000.vrp', 'w') as file:
    file.write("NAME : custom_input_1million\n")
    file.write("COMMENT : \"input file for vrp\"\n")
    file.write("TYPE : CVRP\n")
    file.write("DIMENSION : " + N + "\n")
    file.write("EDGE_WEIGHT_TYPE : EUC_2D\n")
    file.write("CAPACITY : 10000\n")
    file.write("NODE_COORD_SECTION\n")
    count = 1
    for (x, y) in s:
        file.write(str(count) + ' ' + str(x) + ' ' + str(y)+ '\n')
        count += 1
    file.write("DEMAND_SECTION\n")
    for i in range(1, n+1):
        file.write(str(i) + ' ' + str(random.randint(1, 500)) + '\n')
    file.write("DEPOT_SECTION\n")
    file.write("0\n")
    file.write("0\n")
    file.write("EOF")

File: job1436641/gen_inp.cpp
#include <iostream>
#include <set>
#include <bits/stdc++.h>
#include <random>

using namespace std;

double random_uniform(double a, double b) {
    if (a > b) {
        std::swap(a, b); // Swap if a > b
    }

    std::random_device rd;
    std::mt19937 gen(rd());
    std::uniform_real_distribution<> dis(a, b);
    
    return dis(gen);
}

double find_prob(pair<int,int> p1, pair<int,int> p2)
{
    double dist = sqrt(pow(p1.first-p2.first,2) + pow(p1.second-p2.second,2));
    double prob = exp(-dist/20000);
    return prob;
}

const pair<double,double> depot_loc = {0,0};
const int num_customers = 400;
const int num_clusters = 20;
set<pair<int,int>> cluster_points;
priority_queue<pair<double,pair<int,int>>> pq;

int main()
{
    // defining the cluster points
    for(int i=0; i<num_clusters; i++)
    {
        int x = random_uniform(-500,500);
        int y = random_uniform(-500,500);
        while(cluster_points.find({x,y}) != cluster_points.end())
        {
            x = random_uniform(-500,500);
            y = random_uniform(-500,500);
        }
        cluster_points.insert({x,y});
    }

    // check through all integral points in the space
    for(int i=-500; i<=500; i++)
    {
        for(int j=-500; j<=500; j++)
        {
            if(cluster_points.find({i,j}) != cluster_points.end())
                continue;
            double point_prob = 0;
            for(auto cluster_point : cluster_points)
            {
                point_prob += find_prob({i,j},cluster_point);
            }
            pq.push({point_prob,{i,j}});
        }
    }
    // cout << "NAME : inp1.vrp" << endl;
    // cout << "COMMENT : input file for vrp" << endl;
    // cout << "TYPE : CVRP" << endl;
    // cout << "DIMENSION : " << num_customers << endl;
    // cout << "EDGE_WEIGHT_TYPE : EUC_2D" << endl;
    // cout << "CAPACITY : " << 40 << endl;
    // cout << "NODE_COORD_SECTION" << endl;
    int count=0;
    for(auto cluster_point : cluster_points)
    {
        cout << cluster_point.first << "," << cluster_point.second << endl;
        count++;
    }
    for(int i=0; i<num_customers-num_clusters; i++)
    {
        auto point = pq.top();
        pq.pop();
        cout << point.second.first << "," << point.second.second << endl;
        count++;
    }
    // cout << "DEMAND_SECTION" << endl;
    // for(int i=0; i<num_customers; i++)
    // {
    //     int rand_demand = random_uniform(1,100);
    //     cout << i+1 << " " << rand_demand << endl;
    // }
    // cout << "DEPOT_SECTION" << endl;
    // cout << 0 << endl;
    // cout << 0 << endl;
    // cout << "EOF" << endl;
    return 0;
}